### ISSUE TITLE
Using as a client?

### ISSUE THREAD

**Issue by baryluk** [pawelgaczynski/gain#18]
Is there a way to use it as a TCP client? Lets say creating 1000 HTTP output connections sending requests and getting responses.

**Comment by pawelgaczynski** [comment]
Hi. Do you want to create some kind of benchmarking tool?

**Comment by baryluk** [comment]
No, just a client (open connection, do some handshakes) that can connect to many tcp connections, send requests, get respones, etc. Could be anything, benchmarking tools, web crawler, multi connection downloader, web browser (unlikely, just an example), parallel rsync-like functionality, etc. etc.

**Comment by baryluk** [comment]
Another example might QUIC client, or uTP client, or DNS stress testing tool. These are for UDP, but still a client initiated sending. With "connected" UDP socket I am pretty sure it can be done same was as with bind sockets, but for unconnected it is slightly different.
